packages/core-accountsjs/core-accountsjs.js

```diff
@@ -0,0 +1,205 @@
+import { AccountsServer, ServerHooks, AccountsJsError } from '@accounts/server';
+import { AccountsPassword } from '@accounts/password';
+import MongoDBInterface from '@accounts/mongo';
+import { MongoInternals } from 'meteor/mongo';
+import { DatabaseManager } from '@accounts/database-manager';
+import { Random } from 'meteor/random';
+import crypto, { randomBytes } from 'crypto';
+import defer from 'lodash.defer';
+
+// import { destroyToken, hashLoginToken } from './util';
+
+const CreateUserErrors = {
+  /**
+   * Will throw if no username or email is provided.
+   */
+  UsernameOrEmailRequired: 'UsernameOrEmailRequired',
+  /**
+   * Username validation via option `validateUsername` failed.
+   */
+  InvalidUsername: 'InvalidUsername',
+  /**
+   * Email validation via option `validateEmail` failed.
+   */
+  InvalidEmail: 'InvalidEmail',
+  /**
+   * Password validation via option `validatePassword` failed.
+   */
+  InvalidPassword: 'InvalidPassword',
+  /**
+   * Email already exist in the database.
+   */
+  EmailAlreadyExists: 'EmailAlreadyExists',
+  /**
+   * Username already exist in the database.
+   */
+  UsernameAlreadyExists: 'UsernameAlreadyExists',
+};
+
+const METEOR_ID_LENGTH = 17;
+
+export const idProvider = () =>
+  randomBytes(30)
+    .toString('base64')
+    .replace(/[\W_]+/g, '')
+    .substr(0, METEOR_ID_LENGTH);
+
+const dateProvider = (date) => date || new Date();
+
+const mongoStorage = new MongoDBInterface(
+  MongoInternals.defaultRemoteCollectionDriver().mongo.db,
+  {
+    convertUserIdToMongoObjectId: false,
+    convertSessionIdToMongoObjectId: false,
+    idProvider,
+    dateProvider,
+  },
+);
+
+const dbManager = new DatabaseManager({
+  sessionStorage: mongoStorage,
+  userStorage: mongoStorage,
+});
+
+const accountsServerOptions = {
+  siteUrl: Meteor.absoluteUrl(),
```

Comment:
Please don't introduce new code dependencies by using Random or Meteor in this package 😀